build-system/runner/src/org/ampproject/AmpCodingConvention.java

```diff
@@ -35,6 +41,15 @@ public AmpCodingConvention(CodingConvention convention) {
     super(convention);
   }
 
+  @Override public Collection<AssertionFunctionSpec> getAssertionFunctions() {
```

Comment:
I also just found out.

It makes `dev.assert(foo != null)` and similar things participate in type inference, so the compiler knows in the next like that foo is not null.